targets/frontend/src/components/documents/Container.js

```diff
@@ -0,0 +1,164 @@
+import Link from "next/link";
+import { useRouter } from "next/router";
+import PropTypes from "prop-types";
+import { useCallback, useMemo } from "react";
+import { IoMdAdd } from "react-icons/io";
+import { useSelectionContext } from "src/pages/contenus";
+import { Card, Flex, Message } from "theme-ui";
+import { useMutation, useQuery } from "urql";
+
+import { Button } from "../button";
+import { Stack } from "../layout/Stack";
+import { Pagination } from "../pagination";
+import { DocumentsListActions } from "./Actions";
+import { DocumentList } from "./List";
+import { SearchFilters } from "./SearchFilters";
+
+export function DocumentListContainer({ initialFilterValues }) {
+  const router = useRouter();
+  const context = useMemo(() => ({ additionalTypenames: ["documents"] }), []);
+  const [, setSelectedItems] = useSelectionContext();
+  const [, updatePublication] = useMutation(updatePublicationMutation);
+  const updatePublicationStatus = useCallback(
+    (docEntries) => {
+      docEntries.forEach(([cdtnId, isPublished]) => {
+        updatePublication({ cdtnId, isPublished });
+      });
+      setSelectedItems({});
+    },
+    [updatePublication, setSelectedItems]
+  );
+
+  const updateUrl = useCallback(
+    (filterValues) => {
+      // we reset changed published value if search critera change
+      setSelectedItems({});
+      const query = { ...filterValues, page: 0 };
+      router.push({ pathname: router.route, query }, undefined, {
+        shallow: true,
+      });
+    },
+    [router, setSelectedItems]
+  );
+
+  const [result] = useQuery({
```

Comment:
I don't understand how the list is refreshed when we update the publication status of documents. For the search, we push a new path in the router, the component is rerender and useQuery is performed. But when we change the publication status, we call the `updatePublicationStatus` which doesn't trigger a new call to fetch all documents 🤔

Reply:
urql update it's cache since there is a mutation on documents data

Reply:
Ok. (I didn't see that on their documentation, probably miss it 🤷 😕)